### DISCUSSION TITLE
Error installing orbico or mobileracker companion

### DISCUSSION THREAD

**Discussion by derrabe1980** [Guilouz/Creality-Helper-Script-Wiki#558] [Issues]
I just factory reset my k1 max because I was getting a bunch of error after software updates.   So I decided to do a clean install and when I try to install orbico or mobileracker companion I get the follow error and it shutdowns the script

`New python executable in /usr/data/moonraker-obico-env/bin/python3
Also creating executable in /usr/data/moonraker-obico-env/bin/python
Installing setuptools, pip, wheel...done.
  Could not find a version that satisfies the requirement websocket-client==1.2.1 (from -r /usr/data/moonraker-obico/scripts/../requirements.txt (line 1)) (from versions: )
No matching distribution found for websocket-client==1.2.1 (from -r /usr/data/moonraker-obico/scripts/../requirements.txt (line 1))
`

**Comment by outtatime28** [comment]
I am having the same issue with this as well as with OctoEverywhere and OctoApp Companion

**Reply by vgomenyuk** [reply]
The same for me...

**Reply by Schulli73** [reply]
The same for me...

**Comment by Guilouz** [comment]
@kennethjiang Can you check this please

@derrabe1980 My script use installer from Obico, please post issue directly on repo : https://github.com/TheSpaghettiDetective/moonraker-obico

**Comment by Guilouz** [comment]
Obico is fixed. Waiting PR are merged for OctoEveryWhere, OctoApp and Mobileraker.

**Reply by Pokeylooted** [reply]
Obico is still not working, *no module found markupsafe*, this happens during the linking process

**Reply by derrabe1980** [reply]
I was just coming to say orbico also failed for me

**Reply by Guilouz** [reply]
Working for me:

<img width="1334" alt="Capture_decran_2024-05-12_a_21 02 37" src="https://github.com/Guilouz/Creality-Helper-Script-Wiki/assets/12702322/0bdab0bb-11f9-4bcf-9697-39a063ef75ba">

Report the error if needed.

**Reply by Pokeylooted** [reply]
Can you send me what your site-packages folder looks like for the python3.8 in that? For me it keeps installing the wrong version

**Comment by vgomenyuk** [comment]
Link of the printer is failed:[
<img width="898" alt="Screenshot 2024-05-13 at 09 07 51" src="https://github.com/Guilouz/Creality-Helper-Script-Wiki/assets/13455926/21e34671-7595-4bcb-b5d1-a7925c2b51f2">
](url)

**Reply by derrabe1980** [reply]
This is exactly the error I am getting and when I try to run those commands I get directory not found

**Comment by Pokeylooted** [comment]
@Guilouz This can be solved by installing MarkupSafe 1.1.1 . Any version newer than that doesn't play well with python 3.8, so either upgrade python version or downgrade MarkupSafe!

**Reply by Guilouz** [reply]
@Pokeylooted MarkupSafe is supposed to be compatible with Python >=3.7

https://pypi.org/project/MarkupSafe/

**Reply by derrabe1980** [reply]
Still getting the error as before with the directions you provided below

**Comment by Pokeylooted** [comment]
> This is exactly the error I am getting and when I try to run those commands I get directory not found



> Link of the printer is failed:[ <img alt="Screenshot 2024-05-13 at 09 07 51" width="898" src="https://private-user-images.githubusercontent.com/13455926/329940324-21e34671-7595-4bcb-b5d1-a7925c2b51f2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU2MDUxODEsIm5iZiI6MTcxNTYwNDg4MSwicGF0aCI6Ii8xMzQ1NTkyNi8zMjk5NDAzMjQtMjFlMzQ2NzEtNzU5NS00YmNiLWI1ZDEtYTc5MjVjMmI1MWYyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTEzVDEyNTQ0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNmNmI0YTE2OWZiZTE3MjBiZmVmYTk2NjY2ZGVlY2ZiYTNmODliM2U0OGJjOTQ4M2Q2YjUxOWVlMDg5ZjhmMTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.bViNl7y6-XtwvkSnDHIAdxbVTj0dVv4YowxL6mwARAQ"> ](url)

@derrabe1980 @vgomenyuk, this can be solved by running this command: `/usr/data/moonraker-obico-env/bin/python -m pip install MarkupSafe==1.1.1` After which you need to do: `cd /usr/data/moonraker-obico/scripts` , then `sh ./install_creality.sh`

**Reply by derrabe1980** [reply]
![image](https://github.com/Guilouz/Creality-Helper-Script-Wiki/assets/25082366/9aa48c53-600c-4f71-9fd4-a3188639e432)
Same error after trying those steps

**Comment by Pokeylooted** [comment]
> @Pokeylooted MarkupSafe is supposed to be compatible with Python >=3.7
> 
> https://pypi.org/project/MarkupSafe/
@Guilouz, I looked more into it this is the problem that its having. Version 1.1.1 isn't suitable either as 2 of the libraries require 2.0.0 so not sure what we can do to fix this. Look at these errors below *you can get these by upgrading pip that the venv installs*
`Collecting MarkupSafe
  Downloading MarkupSafe-2.1.5.tar.gz (19 kB)
  Preparing metadata (setup.py) ... done
Discarding https://files.pythonhosted.org/packages/87/5b/aae44c6655f3801e81aa3eef09dbbf012431987ba564d7231722f68df02d/MarkupSafe-2.1.5.tar.gz (from https://pypi.org/simple/markupsafe/) (requires-python:>=3.7): Requested MarkupSafe from https://files.pythonhosted.org/packages/87/5b/aae44c6655f3801e81aa3eef09dbbf012431987ba564d7231722f68df02d/MarkupSafe-2.1.5.tar.gz has inconsistent version: expected '2.1.5', but metadata has '0.0.0'
  Downloading MarkupSafe-2.1.4.tar.gz (19 kB)
  Preparing metadata (setup.py) ... done
Discarding https://files.pythonhosted.org/packages/fb/5a/fb1326fe32913e663c8e2d6bdf7cde6f472e51f9c21f0768d9b9080fe7c5/MarkupSafe-2.1.4.tar.gz (from https://pypi.org/simple/markupsafe/) (requires-python:>=3.7): Requested MarkupSafe from https://files.pythonhosted.org/packages/fb/5a/fb1326fe32913e663c8e2d6bdf7cde6f472e51f9c21f0768d9b9080fe7c5/MarkupSafe-2.1.4.tar.gz has inconsistent version: expected '2.1.4', but metadata has '0.0.0'
  Downloading MarkupSafe-2.1.3.tar.gz (19 kB)
  Preparing metadata (setup.py) ... done
Discarding https://files.pythonhosted.org/packages/6d/7c/59a3248f411813f8ccba92a55feaac4bf360d29e2ff05ee7d8e1ef2d7dbf/MarkupSafe-2.1.3.tar.gz (from https://pypi.org/simple/markupsafe/) (requires-python:>=3.7): Requested MarkupSafe from https://files.pythonhosted.org/packages/6d/7c/59a3248f411813f8ccba92a55feaac4bf360d29e2ff05ee7d8e1ef2d7dbf/MarkupSafe-2.1.3.tar.gz has inconsistent version: expected '2.1.3', but metadata has '0.0.0'
  Downloading MarkupSafe-2.1.2.tar.gz (19 kB)
  Preparing metadata (setup.py) ... done
Discarding https://files.pythonhosted.org/packages/95/7e/68018b70268fb4a2a605e2be44ab7b4dd7ce7808adae6c5ef32e34f4b55a/MarkupSafe-2.1.2.tar.gz (from https://pypi.org/simple/markupsafe/) (requires-python:>=3.7): Requested MarkupSafe from https://files.pythonhosted.org/packages/95/7e/68018b70268fb4a2a605e2be44ab7b4dd7ce7808adae6c5ef32e34f4b55a/MarkupSafe-2.1.2.tar.gz has inconsistent version: expected '2.1.2', but metadata has '0.0.0'
  Downloading MarkupSafe-2.1.1.tar.gz (18 kB)
  Preparing metadata (setup.py) ... done
Discarding https://files.pythonhosted.org/packages/1d/97/2288fe498044284f39ab8950703e88abbac2abbdf65524d576157af70556/MarkupSafe-2.1.1.tar.gz (from https://pypi.org/simple/markupsafe/) (requires-python:>=3.7): Requested MarkupSafe from https://files.pythonhosted.org/packages/1d/97/2288fe498044284f39ab8950703e88abbac2abbdf65524d576157af70556/MarkupSafe-2.1.1.tar.gz has inconsistent version: expected '2.1.1', but metadata has '0.0.0'
  Downloading MarkupSafe-2.1.0.tar.gz (18 kB)
  Preparing metadata (setup.py) ... done
Discarding https://files.pythonhosted.org/packages/62/0f/52c009332fdadd484e898dc8f2acca0663c1031b3517070fd34ad9c1b64e/MarkupSafe-2.1.0.tar.gz (from https://pypi.org/simple/markupsafe/) (requires-python:>=3.7): Requested MarkupSafe from https://files.pythonhosted.org/packages/62/0f/52c009332fdadd484e898dc8f2acca0663c1031b3517070fd34ad9c1b64e/MarkupSafe-2.1.0.tar.gz has inconsistent version: expected '2.1.0', but metadata has '0.0.0'
  Downloading MarkupSafe-2.0.1.tar.gz (18 kB)
  Preparing metadata (setup.py) ... done
Discarding https://files.pythonhosted.org/packages/bf/10/ff66fea6d1788c458663a84d88787bae15d45daa16f6b3ef33322a51fc7e/MarkupSafe-2.0.1.tar.gz (from https://pypi.org/simple/markupsafe/) (requires-python:>=3.6): Requested MarkupSafe from https://files.pythonhosted.org/packages/bf/10/ff66fea6d1788c458663a84d88787bae15d45daa16f6b3ef33322a51fc7e/MarkupSafe-2.0.1.tar.gz has inconsistent version: expected '2.0.1', but metadata has '0.0.0'
  Downloading MarkupSafe-2.0.0.tar.gz (18 kB)
  Preparing metadata (setup.py) ... done
Discarding https://files.pythonhosted.org/packages/67/6a/5b3ed5c122e20c33d2562df06faf895a6b91b0a6b96a4626440ffe1d5c8e/MarkupSafe-2.0.0.tar.gz (from https://pypi.org/simple/markupsafe/) (requires-python:>=3.6): Requested MarkupSafe from https://files.pythonhosted.org/packages/67/6a/5b3ed5c122e20c33d2562df06faf895a6b91b0a6b96a4626440ffe1d5c8e/MarkupSafe-2.0.0.tar.gz has inconsistent version: expected '2.0.0', but metadata has '0.0.0'
  Downloading MarkupSafe-1.1.1.tar.gz (19 kB)`

**Reply by Guilouz** [reply]
Yes already made many test, only 1.1.1 version is working no matter the version of pip.

We are on it.

**Reply by Pokeylooted** [reply]
You can fix this by upgrading the Python version to 3.11, it will install it correctly. *You can even manually put it into 3.8, after installing it with 3.11* So maybe upgrading the venv? Or just including that already preinstalled with the script? @Guilouz

**Comment by Pokeylooted** [comment]
> Yes already made many test, only 1.1.1 version is working no matter the version of pip.
> 
> We are on it.

@Guilouz Found a fix, upgrade the python3.8 from 3.8.2 to 3.8.10

**Reply by Guilouz** [reply]
Fixed here: https://github.com/TheSpaghettiDetective/moonraker-obico/pull/99

**Reply by Pokeylooted** [reply]
@Guilouz `./scripts/install_creality.sh: line 128: virtualenv: not found`

**Reply by Guilouz** [reply]
Strange virtualenv is already installed on k1 os

**Comment by vgomenyuk** [comment]
OctoEverywhere, Mobileraker Companion, OctoApp Companion:
```
System package install complete.
Checking Python Virtual Environment For OctoEverywhere...
No virtual environment found, creating one now.
created virtual environment CPython3.11.7.final.0-32 in 14806ms
  creator CPython3Posix(dest=/usr/data/octoeverywhere-env, clear=False, no_vcs_ignore=False, global=True)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/usr/data/.local/share/virtualenv)
    added seed packages: pip==24.0, setuptools==69.5.1, wheel==0.43.0
  activators BashActivator,CShellActivator,FishActivator,NushellActivator,PowerShellActivator,PythonActivator
Updating PIP if needed... (this can take a few seconds or so)
Requirement already satisfied: pip in /usr/data/octoeverywhere-env/lib/python3.11/site-packages (24.0)
Installing or updating required python libs...
  error: subprocess-exited-with-error
      RequiredDependencyException: zlib

      During handling of the above exception, another exception occurred:

      Traceback (most recent call last):
        File "/usr/data/octoeverywhere-env/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/usr/data/octoeverywhere-env/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by steveharman** [comment]
mobileraker installs fine for me.  Obico does not.

```
Configuring python3-sqlite3.
Configuring python3-unittest.
Configuring python3.
Configuring python3-pip.
###### Creating python virtual environment for moonraker-obico...

./scripts/install_creality.sh: line 128: virtualenv: not found
```

I just factory-reset my k1max, upgraded to Creality firmware 1.3.3.29 and began reinstalling all my usual items from Installation Helper.  Everything installed fine except Obico.  Obico is the final item I attempted to install.

The script says its version is 6.0.0

**Reply by Guilouz** [reply]
@steveharman can you try latest script update

**Comment by JDJonny99** [comment]
I have a similar problem. I cannot install Octoeverywhere or remove it. When I tried to remove it and install it afterwards I get the output “already installed”. How can I fix this?

**Reply by Dud3z** [reply]
Same for me

**Reply by mrlordziemniak** [reply]
same here, and it was working on previous version of script, now on 6.2 is not

**Reply by Guilouz** [reply]
There is no change with Octoeverywhere in 6.2 version.

@JDJonny99 What error do you have ?

**Reply by mrlordziemniak** [reply]
I have provlem wirh octoapp plugin, nowni had new firmware on my v3 ke  
Octoapp was working fine on 6.0.0

**Reply by Guilouz** [reply]
Helper script have nothing to see with obico, octoeverywhere , octoapp or mobileraker install, it use the installer provided with them.
Nothing have change in their repo, last change in octoapp is 2weeks ago with my fix for install issue and it working fine.

**Reply by mrlordziemniak** [reply]
Hmm, so i have problem with install,  now I can't uninstall I got error that is nothing to uninstall,  but when I try to install u got error that octoapp is installed,  tomorrow I will try factory reset ,

edit:
after factory reset it is working ;) 

and one question, why on 6.2 all the time before any print printer is calibrating ?

**Reply by kens96** [reply]
I just did a factory reset and installed fluidd, Entware and a few other things. When I got to OctoApp it failed to install anything. I tried to remove and reinstall but I cant. 

> You're about to uninstall OctoApp.
> This printer ID will be deleted, but you can always reinstall the plugin and re-add this printer.
> 
> Are you want to uninstall? [y/n]y
> 
> 
> Starting OctoApp uninstall
> No local plugins or companions were found to remove.
> 
>  ✓ OctoApp Companion has been removed successfully!

but it thinks it is still installed 

>  │ •REMOTE ACCESS:                                              │
>  │   ✗ OctoEverywhere                                           │
>  │   ✗ Obico                                                    │
>  │   ✗ GuppyFLO                                                 │
>  │   ✗ Mobileraker Companion                                    │
>  │   ✓ OctoApp Companion  

Running 6.2.0 but had no issues with 6.0.0 
What should I do please?

**Reply by kens96** [reply]
In the end I just cloned repo and installed manually from /usr/data and plugin is working fine now.
Something not right though with the scripts.
